services/npm-stat/npm-stat-downloads.tester.js

```diff
@@ -0,0 +1,43 @@
+import { isMetricOverTimePeriod, isMetric } from '../test-validators.js'
+import { createServiceTester } from '../tester.js'
+export const t = await createServiceTester()
+
+t.create('weekly downloads of npm author dukeluo')
+  .get('/dw/dukeluo.json')
+  .expectBadge({
+    label: 'downloads',
+    message: isMetricOverTimePeriod,
+    color: 'brightgreen',
+  })
+
+t.create('monthly downloads of npm author dukeluo')
+  .get('/dm/dukeluo.json')
+  .expectBadge({
+    label: 'downloads',
+    message: isMetricOverTimePeriod,
+    color: 'brightgreen',
+  })
+
+t.create('yearly downloads of npm author dukeluo')
+  .get('/dy/dukeluo.json')
+  .expectBadge({
+    label: 'downloads',
+    message: isMetricOverTimePeriod,
+    color: 'brightgreen',
+  })
+
+t.create('total downloads of npm author dukeluo')
+  .get('/dt/dukeluo.json')
+  .expectBadge({
+    label: 'downloads',
+    message: isMetric,
+    color: 'brightgreen',
+  })
+
+t.create('downloads of unknown npm package author')
+  .get('/dt/npm-api-does-not-have-this-package-author.json')
+  .expectBadge({
+    label: 'downloads',
+    message: '0',
+    color: 'red',
+  })
```

Comment:
Does this API allow us to differentiate between a user who does exist with 0 downloads and a non-existant user?
If not, we can roll with this.

Reply:
I think this API cant differentiate between a user who does exist with 0 downloads and a non-existant user. After my testing, this API always return an empty object with a staus code 200 for a non-existant user.